src/test/java/service/impl/ReportGeneratorImplTest.java

```diff
@@ -0,0 +1,77 @@
+package service.impl;
+
+import dao.FruitDao;
+import db.Storage;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import service.ReportGenerator;
+
+class ReportGeneratorImplTest {
+    private static ReportGenerator reportGenerator;
+    private static FruitDao fruitDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        fruitDao = new FruitDao() {
+            @Override
+            public Integer getBalance(String fruit) {
+                return 0;
+            }
+
+            @Override
+            public boolean addBalance(String fruit, int quantity) {
+                return false;
+            }
+
+            @Override
+            public void updateBalance(String fruit, int quantity) {
+
+            }
+
+            @Override
+            public Set<Map.Entry<String, Integer>> getAllEntries() {
+                return Storage.fruitStock.entrySet();
+            }
+        };
+        reportGenerator = new ReportGeneratorImpl(fruitDao);
```

Comment:
why do you implement annonimous class?

Reply:
I used them to make my test class independent from the FruitDaoImpl. Is this wrong?